version: '3.8'

services:
  php-datadog:
    build:
      context: ./docker/php
      dockerfile: Dockerfile.datadog
    environment:
      - DD_SERVICE=symfony-app
      - DD_AGENT_HOST=datadog-agent
      - DD_TRACE_AGENT_PORT=8126
      - DD_API_KEY=${DD_API_KEY}
      - DD_TRACE_DEBUG=true
      - DD_LOGS_ENABLED=true
      - DD_PROCESS_AGENT_ENABLED=true
    volumes:
      - .:/var/www/symfony
      - ./docker/php/.env.sample:/var/www/symfony/.env
    expose:
      - "9000"

  nginx-datadog:
    build:
      context: ./docker/nginx
      dockerfile: Dockerfile.datadog
    ports:
      - "8080:80"
    depends_on:
      - php-datadog
    volumes:
      - ./var/logs/nginx:/var/log/nginx
      - .:/var/www/symfony

  datadog-agent:
    image: datadog/agent:latest
    environment:
      - DD_API_KEY=${DD_API_KEY}
      - DD_SITE=${DD_SITE}
      - DD_APM_ENABLED=true
    ports:
      - "127.0.0.1:8126:8126/tcp"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "/proc/:/host/proc/:ro"
      - "/sys/fs/cgroup/:/host/sys/fs/cgroup:ro"
